src/Parse/ParsePolygon.php

```diff
@@ -38,14 +38,21 @@ public function __construct($coords)
      */
     public function setCoordinates($coords)
     {
-        if (count($coords) < 3 || !is_array($coords)) {
-            throw new ParseException('Polygon must have at least 3 points');
+        if (!is_array($coords)) {
+            throw new ParseException('Coordinates must be an Array');
+        }
+        if (count($coords) < 3) {
+            throw new ParseException('Polygon must have at least 3 GeoPoints or Points');
         }
         $points = [];
         foreach ($coords as $coord) {
-            $geoPoint = $coord;
-            if (is_array($coord)) {
+            $geoPoint;
```

Comment:
Don't really need this here, lexical scoping doesn't apply in this case for php. 

However...with that said I imagine some IDEs would like to highlight later uses as 'potentially undefined', even though the final case is throwing an exception. With that in mind this is okay to leave here.